docs/source/cpp/memory.rst

```diff
@@ -125,3 +125,38 @@ you can do so using the :class:`arrow::stl::allocator` wrapper.
 Conversely, you can also use a STL allocator to allocate Arrow memory,
 using the :class:`arrow::stl::STLMemoryPool` class.  However, this may be less
 performant, as STL allocators don't provide a resizing operation.
+
+Devices
+=======
+
+Many Arrow applications only access CPU (host) memory.  However, in some cases
+it is desired to handle on-device memory (such as on-board memory on a GPU
+add-on card) as well as regular CPU memory.
+
+Arrow represents different devices, including the CPU, using the
+:class:`arrow::Device` abstraction.  Furthermore, one can create
+:class:`arrow::MemoryManager` instances which specifiy how to allocate
+memory on a given device (for example, using a particular
+:class:`arrow::MemoryPool` on the CPU).
+
+Device-Agnostic Programming
+---------------------------
+
+If you receive a Buffer from third-party code, you can query whether it is
+CPU-readable by calling its :func:`~arrow::Buffer::is_cpu` method.
+
+You can also make sure the Buffer can be viewed on a given device, in a
```

Comment:
```suggestion
You can also check generically that a Buffer can be viewed on a given device
```

Reply:
It's not just a check. The Buffer returned by `View` does not have the same address as the original buffer.

Reply:
How about:
```suggestion
You can also generically use a non-eager-copy transfer mechanism to view the Buffer on a
given device, by calling :func:`arrow::Buffer::View` (which will return `nullptr` if only
copy is supported) or :func:`arrow::Buffer::ViewOrCopy` (which will fall back
to copy if :func:`arrow::Buffer::View` would return `nullptr`).
```